src/Build/BuildCheck/API/BuildAnalyzerConfiguration.cs

```diff
@@ -64,32 +65,78 @@ public class BuildAnalyzerConfiguration
     /// </summary>
     /// <param name="configDictionary">The configuration dictionary containing the settings for the build analyzer. The configuration's keys are expected to be in lower case or the EqualityComparer to ignore case.</param>
     /// <returns>A new instance of <see cref="BuildAnalyzerConfiguration"/> with the specified settings.</returns>
-    internal static BuildAnalyzerConfiguration Create(Dictionary<string, string>? configDictionary)
+    internal static BuildAnalyzerConfiguration Create(Dictionary<string, string>? configDictionary) => new()
     {
-        return new()
+        EvaluationAnalysisScope = TryExtractEvaluationAnalysisScope(configDictionary),
+        Severity = TryExtractSeverity(configDictionary),
+    };
+
+
+    private static EvaluationAnalysisScope? TryExtractEvaluationAnalysisScope(Dictionary<string, string>? config)
+    {
+
+        if (!TryExtractValue(BuildCheckConstants.scopeConfigurationKey, config, out string? stringValue) || stringValue is null)
         {
-            EvaluationAnalysisScope = TryExtractValue(nameof(EvaluationAnalysisScope), configDictionary, out EvaluationAnalysisScope evaluationAnalysisScope) ? evaluationAnalysisScope : null,
-            Severity = TryExtractValue(nameof(Severity), configDictionary, out BuildAnalyzerResultSeverity severity) ? severity : null
-        };
+            return null;
+        }
+
+        switch (stringValue)
+        {
+            case "project":
+                return BuildCheck.EvaluationAnalysisScope.ProjectOnly;
+            case "current_imports":
+                return BuildCheck.EvaluationAnalysisScope.ProjectWithImportsFromCurrentWorkTree;
+            case "without_sdks":
+                return BuildCheck.EvaluationAnalysisScope.ProjectWithImportsWithoutSdks;
+            case "all":
+                return BuildCheck.EvaluationAnalysisScope.ProjectWithAllImports;
+            default:
```

Comment:
This is something that yet to be defined: the scope values and the configuration values could  change in the future.

Reply:
@rainersigwald @baronfel Could you please review the update of the configuration of the evaluation scope? 

Another option is: to leave the snake_case values of the evaluation scope `project_only` and update the configuration options once the scope will be changed (in case there will be any changes)

Reply:
@rainersigwald one thing I mentioned to @f-alizada was perhaps a different way of modelling the scopes that's parallel to the NuGet Asset concept. Instead of bespoke scopes like `ProjectOnly`, `ProjectWithImportsFromCurrentWorkTree`, etc. we'd have something more like a `Flags`-style enum, with values like

```cs
[Flags]
enum ImportScope 
{
  Project,
  WorkingDirectory
  MSBuildSDKs
  All = 1 <<<< a billion // catch-all for all flags or whatever
}
```

that could then be composed/declared by users in config via a comma-separated list: `project,working_dir`.

Thoughts on this?

Reply:
Thank you @baronfel, since the evaluation scope parameter and option is not finalized yet, this PR mostly focuses on changing the process of parsing, and decoupling the values defined in the code from the configure options. 
Currently there is no effect how we configure the evaluation scope. 
Overall I do agree with Flags, and the change could be done here If agreed.

Reply:
gotcha - yes parsing could and should be decoupled from the actual values used in the code. if we don't do that then we lose the ability to rename without impacting end-user code!

Reply:
Per discssion. 
The updates of the Scope values will be updated once the definition of scopes will be defined. 
Resolving the comment, thank you for the input!